File: src/main/java/com/ideabobo/constant/CouponState.java
package com.ideabobo.constant;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

/**
 * 用户持有券状态（fs_dict.coupon_state；模板券 typeid=1 通常无状态）
 */
public final class CouponState {

    public static final int NORMAL = 1;
    public static final int USED = 2;

    private static final Map<Integer, String> CODE_TO_LABEL;
    private static final Map<String, Integer> LABEL_TO_CODE;

    static {
        Map<Integer, String> c2l = new HashMap<>();
        c2l.put(NORMAL, "正常");
        c2l.put(USED, "已使用");
        CODE_TO_LABEL = Collections.unmodifiableMap(c2l);

        Map<String, Integer> l2c = new HashMap<>();
        for (Map.Entry<Integer, String> e : c2l.entrySet()) {
            l2c.put(e.getValue(), e.getKey());
        }
        LABEL_TO_CODE = Collections.unmodifiableMap(l2c);
    }

    private CouponState() {
    }

    public static String labelOf(Integer code) {
        if (code == null || code <= 0) {
            return "";
        }
        String label = CODE_TO_LABEL.get(code);
        return label != null ? label : "";
    }

    public static Integer codeOfLabel(String label) {
        if (label == null) {
            return null;
        }
        String t = label.trim();
        if (t.isEmpty()) {
            return null;
        }
        Integer exact = LABEL_TO_CODE.get(t);
        if (exact != null) {
            return exact;
        }
        if (t.contains("已使用")) {
            return USED;
        }
        if (t.contains("正常")) {
            return NORMAL;
        }
        return null;
    }
}

File: app/common/couponState.js
/** 用户优惠券状态，与 fs_dict.coupon_state、后端 CouponState 一致 */
export const COUPON_STATE = {
  NORMAL: 1,
  USED: 2
}

const LABEL_TO_CODE = {
  '正常': COUPON_STATE.NORMAL,
  '已使用': COUPON_STATE.USED
}

const CODE_TO_LABEL = {}
Object.keys(LABEL_TO_CODE).forEach((k) => {
  CODE_TO_LABEL[LABEL_TO_CODE[k]] = k
})

export function couponStateCode(label) {
  if (label == null || label === '') return null
  const key = String(label).trim()
  return LABEL_TO_CODE[key] != null ? LABEL_TO_CODE[key] : null
}

export function couponStateLabel(code) {
  if (code == null) return ''
  return CODE_TO_LABEL[Number(code)] || ''
}

export function isCouponNormal(row) {
  const st = row && row.state != null ? Number(row.state) : couponStateCode(row && row.statecn)
  return st === COUPON_STATE.NORMAL
}
